### PR TITLE
Remove unused "themis_message_*" functions

### PR DESCRIPTION
**Pull request by ilammy**
It turns out these functions are not used by any other code, they were not reviewed or touched in a long time, they are not exported, and they contain a bunch of issues which were never discovered because the code was not used. Remove the dead code.

This has been discovered by @luismerino  in #712.

## Checklist

- [X] Change is covered by automated tests
- [X] Changelog is updated

### CODE REVIEW

**Review by Lagovas — APPROVED** [review]

**Review by iamnotacake — COMMENTED** [review]
If your changes to `Themis.xcodeproj` are also covered by automated tests, then I see no problem here

**Comment by ilammy** [comment]
@iamnotacake,

> If your changes to Themis.xcodeproj are also covered by automated tests, then I see no problem here

Yes, they are. `.github/workflows/test-objc.yaml` builds ObjCThemis from that project file for Carthage, and Bitrise does that as well.

**Comment by vixentael** [comment]
Do we have anymore scripts that add these files into searchable headers?

**Comment by ilammy** [comment]
> Do we have anymore scripts that add these files into searchable headers?

I don't think so. Other places use `src/themis/*.h` whenever they install things.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ _Code:_
   - Include embedded BoringSSL into Soter for convenience ([#681](https://github.com/cossacklabs/themis/pull/681), [#702](https://github.com/cossacklabs/themis/pull/702)).
   - `make deb` and `make rpm` with `ENGINE=boringssl` will now produce `libthemis-boringssl` packages with embedded BoringSSL ([#683](https://github.com/cossacklabs/themis/pull/683), [#686](https://github.com/cossacklabs/themis/pull/686)).
   - `secure_session_create()` now allows only EC keys, returning an error for RSA ([#693](https://github.com/cossacklabs/themis/pull/693)).
+  - Cleaned up unused private API ([#714](https://github.com/cossacklabs/themis/pull/714)).
```
Comment (vixentael) [inline CHANGELOG.md:14]:
```suggestion
  - Cleaned up unused private API ([#714](https://github.com/cossacklabs/themis/pull/714)) Thanks to @luismerino for pointing out.
```

**Review by vixentael — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ilammy) [reply CHANGELOG.md:14]:
```suggestion
  - Cleaned up unused private API. Thanks to [**@luismerino**](https://github.com/luismerino) for pointing this out ([#714](https://github.com/cossacklabs/themis/pull/714)).
```

**Comment by vixentael** [comment]
Wonderful!

**Pull request merged**